src/mpi/romio/adio/common/ad_iread_fake.c

```diff
@@ -17,24 +17,19 @@ void ADIOI_FAKE_IreadContig(ADIO_File fd, void *buf, int count,
 {
     ADIO_Status status;
     MPI_Count typesize;
-    MPI_Offset len;
-    int actual_len = 0;
-
-
-    MPI_Type_size_x(datatype, &typesize);
-    len = (MPI_Offset) count *(MPI_Offset) typesize;
+    int read_count;
+    MPI_Offset nbytes = 0;
 
     /* Call the blocking function.  It will create an error code
      * if necessary.
      */
-    ADIOI_Assert(len == (int) len);     /* the count is an int parm */
-    ADIO_ReadContig(fd, buf, (int) len, MPI_BYTE, file_ptr_type, offset, &status, error_code);
+    ADIO_ReadContig(fd, buf, count, datatype, file_ptr_type, offset, &status, error_code);
     if (*error_code != MPI_SUCCESS) {
-        len = 0;
-    } else {
-        MPI_Get_count(&status, MPI_BYTE, &actual_len);
+        MPI_Type_size_x(datatype, &typesize);
+        MPI_Get_count(&status, datatype, &read_count);
+        nbytes = read_count * typesize;
```

Comment:
Does this mean that we only get count when `*error_code != MPI_SUCCESS` (instead of when it's successful)?

I think having both `if-` and `else-` branches are easier to read.

Reply:
In all cases `nbytes` is initialized to zero, but sure I think putting that in the else branch is probably easier to understand. Will force push a change.